simulator-spring-boot/src/test/java/org/citrusframework/simulator/scenario/SimulatorScenarioTest.java

```diff
@@ -67,7 +67,7 @@ void withRandomTestCaseRunnerImplementation() {
 
             fixture.registerException(cause);
 
-            verify(scenarioEndpointMock).fail(cause);
+            verifyNoInteractions(scenarioEndpointMock);
```

Comment:
Have you thought this over? I stumbled over this in the implementation above already. This pretty much changes the runtime behaviour of scenario execution.

Reply:
I've refactored this part, so the existing API stays in place for now.